src/components/KCopy/KCopy.vue

```diff
@@ -0,0 +1,263 @@
+import { KUI_ICON_SIZE_40 } from '@kong/design-tokens'
+
+const props = defineProps({
+  /**
+    * Text displayed before the copyable text when
+    * `badge` is true
+    */
+  badgeText: {
+    type: String,
+    default: '',
+  },
+  /**
+    * The copyable text
+    */
+  text: {
+    type: String,
+    required: true,
+  },
+  /**
+    * Tooltip text displayed on hover over the `text`
+    */
+  textTooltip: {
+    type: String,
+    default: '',
+  },
+  /**
+    * Tooltip text displayed on hover over copy button
+    */
+  copyTooltip: {
+    type: String,
+    default: 'Copy',
+  },
+  /**
+    * Formatting for copyable text (default, hidden, redacted, deleted)
+    */
+  format: {
+    type: String as PropType<'default' | 'hidden' | 'redacted' | 'deleted'>,
+    required: false,
+    default: 'default',
+    validator: (val: string) => ['default', 'hidden', 'redacted', 'deleted'].includes(val),
+  },
+  /**
+    * Whether or not to display as a badge
+    */
+  badge: {
+    type: Boolean,
+    default: false,
+  },
+  /**
+    * False if `badge` is true
+    */
+  monospace: {
+    type: Boolean,
+    default: true,
+  },
+  /**
+    * Whether or not the text should be truncated
+    */
+  truncate: {
+    type: Boolean,
+    default: false,
+  },
+  /**
+    * Tooltip text displayed on successful copy
+    */
+  successTooltip: {
+    type: String,
+    default: 'Copied',
+  },
+  /**
+    * Number of characters to truncate at
+    */
+  truncationLimit: {
+    type: Number,
+    default: 8,
+  },
+})
+
+const emit = defineEmits<{
+  (e: 'copy', val: string): void
+}>()
+
+const hasSuccessTooltip = computed((): boolean => !!(props.copyTooltip && props.successTooltip))
+const tooltipText = ref(props.copyTooltip)
+
+const badgeTextLabel = computed((): string => {
+  if (!props.badgeText) {
+    return ''
+  }
+
+  // strip trailing colon from label if one exists
+  return props.badgeText.replace(/:$/, '')
+})
+
+const iconTitle = computed(() => {
+  if (props.badgeText) {
+    return `Copy ${badgeTextLabel.value}`
+  }
+  return 'Copy'
+})
+
+const truncateLimitText = computed((): string | null => props.truncate ? `${props.text.substring(0, props.truncationLimit) + '...'}` : null)
+
+// Computed for all dynamic classes
+const truncateStyles = computed((): string | null => props.truncate ? 'copy-element' : null)
+const badgeStyles = computed((): string | null => props.badge ? 'badge-styles' : null)
+const truncateContent = computed((): string | null => props.truncate ? 'truncate-content' : null)
+const useMono = computed((): string | null => props.badge ? null : 'monospace')
+
+const textFormat = computed(() => {
+  if (props.format === 'redacted') {
+    return '*****'
+  } else if (props.format === 'deleted') {
+    return `*${props.text.substring(0, 5)}`
+  }
+  return (props.truncate && props.truncationLimit && truncateLimitText.value) ? truncateLimitText.value.replace(/^"(.*)"$/, '$1') : props.text
```

Comment:
suggestion: can you leave a comment here explaining the return logic?

Reply:
added